README.md

```diff
@@ -14,80 +17,89 @@ The formatter turns code like this:
 
 ```dart
 // BEFORE formatting
-if (tag=='style'||tag=='script'&&(type==null||type == TYPE_JS
-      ||type==TYPE_DART)||
-  tag=='link'&&(rel=='stylesheet'||rel=='import')) {}
+process = await Process.start(path.join(p.pubCacheBinPath,Platform.isWindows
+?'${command.first}.bat':command.first,),[...command.sublist(1),'web:0',
+// Allow for binding to a random available port.
+],workingDirectory:workingDir,environment:{'PUB_CACHE':p.pubCachePath,'PATH':
+path.dirname(Platform.resolvedExecutable)+(Platform.isWindows?';':':')+
+Platform.environment['PATH']!,},);
 ```
 
 into:
 
 ```dart
 // AFTER formatting
-if (tag == 'style' ||
-  tag == 'script' &&
-      (type == null || type == TYPE_JS || type == TYPE_DART) ||
-  tag == 'link' && (rel == 'stylesheet' || rel == 'import')) {}
+process = await Process.start(
+  path.join(
+    p.pubCacheBinPath,
+    Platform.isWindows ? '${command.first}.bat' : command.first,
+  ),
+  [
+    ...command.sublist(1), 'web:0',
+    // Allow for binding to a random available port.
+  ],
+  workingDirectory: workingDir,
+  environment: {
+    'PUB_CACHE': p.pubCachePath,
+    'PATH':
+        path.dirname(Platform.resolvedExecutable) +
+        (Platform.isWindows ? ';' : ':') +
+        Platform.environment['PATH']!,
+  },
+);
 ```
 
 The formatter will never break your code&mdash;you can safely invoke it
 automatically from build and presubmit scripts.
 
-## Using the formatter
+## Formatting files
 
 The formatter is part of the unified [`dart`][] developer tool included in the
-Dart SDK, so most users get it directly from there. That has the latest version
-of the formatter that was available when the SDK was released.
+Dart SDK, so most users run it directly from there using `dart format`.
 
 [`dart`]: https://dart.dev/tools/dart-tool
 
 IDEs and editors that support Dart usually provide easy ways to run the
-formatter. For example, in WebStorm you can right-click a .dart file and then
-choose **Reformat with Dart Style**.
+formatter. For example, in Visual Studio Code, formatting Dart code will use
+the dart_style formatter by default. Most users have it set to reformat every
+time they save a file.
 
 Here's a simple example of using the formatter on the command line:
 
-    $ dart format test.dart
+```sh
+$ dart format test.dart
```

Comment:
[very nitpicky] WDYT about `dart format foo.dart`? `test.dart` is not a file anyone should write since the convention is `foo_test.dart`.

Reply:
Oh, good point. Done (more or less).